plugins/GSdx/Window/GSWndEGL.cpp

```diff
@@ -333,10 +331,18 @@ void *GSWndEGL_X11::CreateNativeWindow(int w, int h)
 	return (void*)&m_NativeWindow;
 }
 
-void *GSWndEGL_X11::AttachNativeWindow(void *handle)
+void GSWndEGL_X11::AttachNativeWindow(void *handle, void **out_native_display, void **out_native_window)
 {
-	m_NativeWindow = *(Window*)handle;
-	return handle;
+	uptr *window_ptr = (uptr*)handle + 1;
+	m_NativeWindow = (Window)*window_ptr;
+
+	// We only need a native display when we manage the window ourself.
+	// By default, EGL will create its own native display. This way the driver knows
+	// that display will be thread safe and so it can enable multithread optimization.
```

Comment:
This comment was related to the scope of CreateNativeDisplay and doesn't make much more sense anymore

Reply:
I still think it's relevant to explain why we use `nullptr` here when we have a perfectly valid X `Display` passed to us through `pDsp` (aka `handle`).

Reply:
Right, makes sense, maybe not the best place to have it but I won't bother you with that